keras/metrics.py

```diff
@@ -1681,6 +1911,8 @@ def clone_metrics(metrics):
 mape = MAPE = mean_absolute_percentage_error
 msle = MSLE = mean_squared_logarithmic_error
 cosine = cosine_similarity = cosine_proximity
+if K.backend() == 'tensorflow':
+    MeanIoU = K.MeanIoU
```

Comment:
Does that actually work with the Keras `compile`/`fit`/`evaluate` methods? If I recall correctly, there were changes made between tf.keras and external Keras (e.g. the `update_state` method returning a list).

Reply:
Related: please add a metric test that does, roughly, for every `M` among the object metrics:

```
model = Sequential([Dense(2, input_shape=(3,)])
model.compile('rmsprop', metrics=[M()])
model.fit(x, y)
loss_val, metric_val = model.evaluate(x, y)
```
No need to test correctness (which is tested separately).

Importantly this test does not need to exclude Theano and CNTK (so it should be in a separate file). Theano / CNTK should only be excluded specifically for MeanIoU.

Reply:
Yes it works with Keras compile/fit/evaluate but i have made the changes as suggested before for raising the error messages and also added a test file.